src/org/zaproxy/zap/extension/scripts/ExtenderScriptHelper.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ *
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ * Copyright 2017 The ZAP development team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.zap.extension.scripts;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.parosproxy.paros.view.View;
+import org.zaproxy.zap.extension.api.API;
+
+public class ExtenderScriptHelper {
```

Comment:
This class should behave like `ExtensionHook`, it would make the scripts less error prone (i.e. no need to manually register and uninstall (the most common) components).